packages/node-app/src/tests/giveaway.controller.test.ts

```diff
@@ -254,3 +256,76 @@ describe('GET /giveaways/:id/generate-winners', () => {
     );
   });
 });
+
+describe('PUT giveaways/:id/participants/:participantId', () => {
+  const participant = {
+    id: 'user',
+    state: ParticipantState.PENDING,
+  };
+
+  const giveawayData = {
+    title: 'Giveaway',
+    description: 'Description for giveaway',
+    startTime: Date.now() + 10000,
+    endTime: Date.now() + 12000,
+    numberOfWinners: 1,
+    prize: 'Test prize',
+    image: 'test-image-base64',
+    participants: [participant],
+  };
+
+  it('should fail if invalid participant', async () => {
+    const giveaway = await Giveaway.create(giveawayData);
+
+    const body = { state: 'invalid' };
+    const res = await request(app)
+      .put(`/giveaways/${giveaway._id}/participants/invalid`)
+      .send(body);
+
+    expect(res.status === 400);
```

Comment:
```suggestion
    const res = await request(app)
      .put(`/giveaways/${giveaway._id}/participants/invalid`)
      .send(body)
      .expect(400);
```

Reply:
The goal here is to divide clearly the steps of each test:

```
setup

action to be tested

assertions
```

Reply:
In that case let's use `expect(res.status).toEqual(400)` since it will give us better error messages in case it fails, making it easier to fix tests